src/MICore/LaunchOptions.cs

```diff
@@ -172,6 +172,39 @@ public EnvironmentEntry(Xml.LaunchOptions.EnvironmentEntry xmlEntry)
         public string Value { get; private set; }
     }
 
+    public sealed class PathMapEntry
+    {
+        public PathMapEntry()
+        {
+        }
+
+        public PathMapEntry(Xml.LaunchOptions.PathMapEntry xmlEntry)
+        {
+            if (string.IsNullOrEmpty(xmlEntry.EditorPath))
+            {
+                throw new ArgumentNullException("EditorPath");
+            }
+            this.EditorPath = xmlEntry.EditorPath;
+            this.CompilerPath = xmlEntry.CompilerPath == null ? string.Empty : xmlEntry.CompilerPath;
+            this.UseForBreakpoints = xmlEntry.UseForBreakpoints;
+        }
+
+        public string EditorPath { get; set; }
```

Comment:
I like Pierson's idea of 'CompileTimePath'. I personally think 'CurrentPath' could be confusing in remote debugging cases. Should I ask around the team room and see if people can understand?

---
In reply to: [98105505](https://github.com/Microsoft/MIEngine/pull/541#discussion_r98105505) [](ancestors = 98105505)